### DISCUSSION TITLE
Simplest _complete_ plugin file for setting python_host_prog

### DISCUSSION THREAD

**Discussion by pjgoodall** [LazyVim/LazyVim#1893] [Q&A]
I'm a beginner with Neovim, LazyVim, and Lua.

Running `:checkHealth` gives me the error `g:python3_host_prog is not set`. It does find my various Python instances, but I want a deterministic result.

I have a Conda Python environment installed to support Neovim, so that it can get its Python needs directly from there - regardless of the virtual environment it is running in:

```bash
❯ conda activate pynvim
❯ which python
/home/<me>/miniconda3/envs/pynvim/bin/python
```

The advice on setting the `python3_host_prog` is baffling. The Neovim help provides suggestions in what I think is Vim script, though there are no nearby instructions as to where that would be applied:
```
let g:python3_host_prog = '/path/to/python3'
```

Other sources give a one-liner that results in an 'Expected a table' error if I try them as a `plugins/python.lua` file:

```lua
-- Specify the Python provider
vim.g.python_host_prog = '/path/to/python'
```

I think that's because the `python_host_prog` doesn't exist yet in `vim.g`.

Unfortunately, all the answers from my searches don't provide instructions at a level I can understand. 

I believe I need help with the correct incantation to add to `lazy.lua`, or more cleanly as a stand-alone `plugins/python.lua`.

Any help would be much appreciated...

**Comment by folke** [comment]
You need to add `vim.g.python_host_prog = '/path/to/python'` to your `options.lua`

**Reply by pjgoodall** [reply]
Thanks for your help - that snippet didn't cause an error but did not seem to fix the diagnostic: 

``` txt
Python 3 provider (optional) ~
- `g:python3_host_prog` is not set.  Searching for python3 in the environment.
- Multiple python3 executables found.  Set `g:python3_host_prog` to avoid surprises.
- Executable: /home/<me>/miniconda3/bin/python3
- Other python executable: /usr/bin/python3
- Other python executable: /bin/python3
- Python version: 3.8.17
- pynvim version: 0.4.3
- OK Latest pynvim is installed.

Python virtualenv ~
- OK no $VIRTUAL_ENV
```

I tried `vim.g.python3_host_prog = '/path/to/python'` , but got a different more elaborate error. I will look into that tomorrow.

**Reply by somu-code** [reply]
@folke Thanks for the solution.
This solution works in my case. @pjgoodall You might need to setup virtual environment for pynivm.
